src/scales/categoryScale.ts

```diff
@@ -3,7 +3,6 @@
 module Plottable {
 export module Scales {
   export class Category extends Scale<string, number> {
-    protected _d3Scale: D3.Scale.OrdinalScale;
```

Comment:
So, unless I'm mistaken the technique that is used here is called "shadowing" where `this._d3Scale` in this context shares the same reference as the one in the parent class.

One problem I have with this technique is that if the parent class were to change it's instance variable name, we can easily enter a scenario where the child class's shadowed instance variable is accidentally not also renamed and thus we have 2 instance variables when we should have one.

To give an example, I think this has caused a bug in the past that was really hard to detect.  The `_isVertical` flag I believe used to be named `isVertical`.  However, for some reason it was used in quite a few places in the hierarchy.  Someone (probably I) changed `isVertical` to `_isVertical` and forgot to change this fact on some of the child classes.  Thankfully I think we wrote our code such that we didn't run into user-facing bugs, but this sounds like something that could easily be dangerous in a quiet way which is quite scary to me.

What do you think?